src/server/server_family.cc

```diff
@@ -2831,10 +2831,16 @@ void ServerFamily::ReplicaOfInternal(CmdArgList args, Transaction* tx, SinkReply
   // If the replication attempt failed, clean up global state. The replica should have stopped
   // internally.
   util::fb2::LockGuard lk(replicaof_mu_);  // Only one REPLICAOF command can run at a time
-  if (ec && replica_ == new_replica) {
-    service_.SwitchState(GlobalState::LOADING, GlobalState::ACTIVE);
-    SetMasterFlagOnAllThreads(true);
-    replica_.reset();
+  // If there was an error above during Start we must not start the main replication fiber.
+  // However, it could be the case that Start() above connected succefully and by the time
+  // we acquire the lock, the context got cancelled because another ReplicaOf command
+  // executed and acquired the replicaof_mu_ before us.
+  if (ec || new_replica->IsContextCancelled()) {
```

Comment:
Small optimization. We do not need to start the replication fiber if we got an error or if the context is cancelled. 

This would be handled anyway by `new_replica->StartMainReplicationFiber` which would spawn the MainReplicationFiber and then exit because it got cancelled. But yeah, no reason to do that